fireplace/game.py

```diff
@@ -112,7 +112,9 @@ def action_block(self, source, actions, type, index=-1, target=None, event_args=
 	def attack(self, source, target):
 		type = BlockType.ATTACK
 		actions = [Attack(source, target)]
-		return self.action_block(source, actions, type, target=target)
+		result = self.action_block(source, actions, type, target=target)
+		if self.state != State.COMPLETE:
+			self.manager.step(Step.MAIN_ACTION, Step.MAIN_END)
```

Comment:
Looks like I forgot to do 'return result' although the results are never used anyway I think.

When the attack block unravels, deaths are processed. This can cause the game to end. If the game is end, the step is already changed to FINAL_WRAPUP so we don't touch it. Otherwise we change it back to MAIN_ACTION because it's still the current player's turn.